### PR TITLE
Chore/more cleanup and housekeeping

### PR DESCRIPTION
**Pull request by tsubik**
- updating gems just conservative update for now
- removing not needed gems
- fix DB queries when loading active admin application code, filter's collection should be proc if it makes db query or is dynamic
- upgrade ruby version
- low-hanging fruit of extra specs for admin panel (bringing up code coverage from 63% to 77%)

### CODE REVIEW

**Comment by tsubik** [comment]
I decided to remove all depredations in this PR as well.

**Comment by tsubik** [comment]
Or actually no, first we need to add couple more specs to bring up coverage to at least 80, 85%

File: app/admin/operator_document_history.rb [hunk 1]
```diff
@@ -65,7 +62,7 @@
     column :operator_document_created_at
     column :uploaded_by
     column :attachment do |o|
-      o.attachment&.filename
+      o&.document_file&.attachment&.filename
```
Comment (santostiago) [inline app/admin/operator_document_history.rb:65]: this was working? 😮

File: spec/controllers/active_admin_spec.rb [hunk 2]
```diff
@@ -15,22 +16,34 @@
 
     if resource.is_a?(ActiveAdmin::Page) || resource.defined_actions.include?(:index)
       describe "GET index" do
-        it "returns http success" do
-          get :index
+        subject { get :index }
```
Comment (santostiago) [inline spec/controllers/active_admin_spec.rb:19]: ❤️ much better.

File: spec/fixtures/snapshots/importers/geojson_fmus_importer.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        }
+      }
+    },
+    "errors": {}
+  },
+  "2": {
+    "attributes": {},
+    "errors": {
+      "record": [
+        "record with 100188 id is not exists"
```
Comment (santostiago) [inline spec/fixtures/snapshots/importers/geojson_fmus_importer.json:138]: I'm not sure why the English here is so bad...

File: spec/importers/geojson_fmus_importer_spec.rb [hunk 4]
```diff
@@ -4,24 +4,32 @@
...
     it "returns right result" do
       importer.import
-      expect(importer.results.as_json).to eq(JSON.parse(results))
+
+      expect(importer.results.to_json).to match_snapshot(
+        "importers/geojson_fmus_importer",
+        # because of the problem on CI that ogr2 returns dates in different formats
+        # and I'm not sure how to restrict format using that tool
+        # in that case I would just ignore those properties by treating them as dynamic ones
+        dynamic_attributes: %w[end_date last_updat start_date]
```
Comment (santostiago) [inline spec/importers/geojson_fmus_importer_spec.rb:31]:
```suggestion
        dynamic_attributes: %w[end_date last_update start_date]
```

File: spec/importers/observations_importer_spec.rb [hunk 5]
```diff
@@ -5,13 +5,14 @@
...
-  let(:results) { fixture_file_upload("#{importer_type}/results.json", Mime[:json].to_s).read }
   let(:importer) { FileDataImport::BaseImporter.build(importer_type, uploaded_file) }
 
   context "CSV" do
     it "returns right result" do
       importer.import
-      expect(importer.results.as_json).to eq(JSON.parse(results))
+
+      expect(importer.results.to_json).to match_snapshot("importers/observations_importer")
+      # expect(importer.results.as_json).to eq(JSON.parse(results))
```
Comment (santostiago) [inline spec/importers/observations_importer_spec.rb:15]: Forgot a comment line.

**Review by santostiago — APPROVED** [review]
Codewise it looks good! (just some minor things) 👍 
Should I also run a smoke test on the branch?

File: app/admin/operator_document_history.rb (hunk 1 above)
Comment (tsubik) [reply app/admin/operator_document_history.rb:65]: No it was not, the csv was failing, looks like noone needed it.

File: spec/fixtures/snapshots/importers/geojson_fmus_importer.json (hunk 3 above)
Comment (tsubik) [reply spec/fixtures/snapshots/importers/geojson_fmus_importer.json:138]: haha, I did not noticed :D

File: spec/importers/geojson_fmus_importer_spec.rb (hunk 4 above)
Comment (tsubik) [reply spec/importers/geojson_fmus_importer_spec.rb:31]: that is the exact name of that date exported from shapefile ;] this is just to not error on wrong format as I described in the comment.

File: spec/importers/observations_importer_spec.rb (hunk 5 above)
Comment (tsubik) [reply spec/importers/observations_importer_spec.rb:15]: ahh, thanks

File: spec/importers/geojson_fmus_importer_spec.rb (hunk 4 above)
Comment (DimaZab) [reply spec/importers/geojson_fmus_importer_spec.rb:31]: [File raster formats, other than GRID, are supported through a dBase file (.vat, .dbf). dBase table field names have a ten-character limit, so the raster attribute tables also have a ten-character limit.](https://support.esri.com/en/technical-article/000008662)

Comment (tsubik) [reply spec/importers/geojson_fmus_importer_spec.rb:31]: Thanks for explanation :)

**Pull request merged**